ksqldb-engine/src/main/java/io/confluent/ksql/schema/ksql/inference/SchemaRegisterInjector.java

```diff
@@ -76,27 +78,42 @@ private void registerForCreateSource(final ConfiguredStatement<? extends CreateS
     // since this injector is chained after the TopicCreateInjector,
     // we can assume that the kafka topic is always present in the
     // statement properties
+    final CreateSource statement = cs.getStatement();
+    final LogicalSchema schema = statement.getElements().toLogicalSchema();
 
-    final LogicalSchema schema = cs.getStatement().getElements().toLogicalSchema();
-
-    final FormatInfo valueFormat = SourcePropertiesUtil
-        .getValueFormat(cs.getStatement().getProperties());
+    final FormatInfo keyFormat = SourcePropertiesUtil.getKeyFormat(statement.getProperties());
+    final SerdeFeatures keyFeatures = SerdeFeaturesFactory.buildKeyFeatures(
+        schema,
+        FormatFactory.of(keyFormat)
+    );
+    registerSchema(
+        schema.key(),
+        statement.getProperties().getKafkaTopic(),
+        keyFormat,
+        keyFeatures,
+        cs.getSessionConfig().getConfig(false),
+        cs.getStatementText(),
+        false,
+        KsqlConstants.SCHEMA_REGISTRY_KEY_SUFFIX
+    );
 
+    final FormatInfo valueFormat = SourcePropertiesUtil.getValueFormat(statement.getProperties());
     final SerdeFeatures valFeatures = SerdeFeaturesFactory.buildValueFeatures(
         schema,
         FormatFactory.of(valueFormat),
-        cs.getStatement().getProperties().getValueSerdeFeatures(),
+        statement.getProperties().getValueSerdeFeatures(),
         cs.getSessionConfig().getConfig(false)
     );
 
     registerSchema(
-        schema,
-        cs.getStatement().getProperties().getKafkaTopic(),
+        schema.withoutPseudoAndKeyColsInValue().value(),
```

Comment:
nit: not your code but can we add a quick unit test to ensure that `withoutPseudoAndKeyColsInValue()` here isn't lost? There doesn't appear to be one right now.

Reply:
This might actually not be needed. Be interesting to see if this can be removed.

Reply:
I confirmed anecdotally and also by following the code path that it isn't necessary. The statement as passed in as a statement (as written) and the logical schema is derived from the `getElements` of the statement. 

Also `withPseudoAndKeyColsInValue` is only called from `LogicalPlanner`, `Analyzer` and `SourceBuilder`, all of which happen after the injector phase.